.github/workflows/test-integrations-windows.yml

```diff
@@ -0,0 +1,1201 @@
+  GOPRIVATE: github.com/hashicorp # Required for enterprise deps
+
+jobs:
+  setup:
+    runs-on: ubuntu-latest
+    name: Setup
+    outputs:
+      compute-small: ${{ steps.runners.outputs.compute-small }}
+      compute-medium: ${{ steps.runners.outputs.compute-medium }}
+      compute-large: ${{ steps.runners.outputs.compute-large }}
+      compute-xl: ${{ steps.runners.outputs.compute-xl }}
+      enterprise: ${{ steps.runners.outputs.enterprise }}
+    steps:
+      - uses: actions/checkout@8e5e7e5ab8b370d6c329ec480221332ada57f0ab # v3.5.2
+      - id: runners
+        run: .github/scripts/get_runner_classes_windows.sh
+
+  dev-build:
+    uses: ./.github/workflows/reusable-dev-build-windows.yml
+    with:
+      runs-on: ${{ needs.setup.outputs.compute-xl }}
+      repository-name: ${{ github.repository }}
+      uploaded-binary-name: 'consul.exe'
+    secrets:
+      elevated-github-token: ${{ secrets.ELEVATED_GITHUB_TOKEN }}
+
+  envoy-integration-test:
+    needs:
+      - setup
+      - dev-build
+    runs-on: ${{ fromJSON(needs.setup.outputs.compute-large) }}
+    permissions:
+      id-token: write # NOTE: this permission is explicitly required for Vault auth.
+      contents: read
+    strategy:
+      fail-fast: false
+      matrix:
+        envoy-version: [ "1.23.10", "1.24.8", "1.25.7", "1.26.2" ]
+        xds-target: [ "server", "client" ]
+    env:
+      ENVOY_VERSION: ${{ matrix.envoy-version }}
+      XDS_TARGET: ${{ matrix.xds-target }}
+      AWS_LAMBDA_REGION: us-west-2
+    steps:
+      - uses: actions/checkout@8e5e7e5ab8b370d6c329ec480221332ada57f0ab # v3.5.2
+      - uses: actions/setup-go@fac708d6674e30b6ba41289acaab6d4b75aa0753 # v4.0.1
+        with:
+          go-version-file: 'go.mod'
+
+      - name: Fetch binary
+        uses: actions/download-artifact@9bc31d5ccc31df68ecc42ccf4149144866c47d8a # v3.0.2
+        with:
+          name: '${{ env.CONSUL_BINARY_UPLOAD_NAME }}'
+          path: ${{ github.workspace }}
+
+      - name: Restore mode+x
+        run: chmod +x ${{ github.workspace }}\consul.exe
+
+      - name: Setup TcpDump Docker Image
+        shell: bash
+        run: |
+          cd test/integration/connect/envoy
+          curl -sSL "https://asheshvidyut-bucket.s3.ap-southeast-2.amazonaws.com/tcpdump.exe" -o tcpdump.exe
+          docker build -t envoy-tcpdump -f Dockerfile-tcpdump-windows .
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@4b4e9c3e2d4531116a6f8ba8e71fc6e2cb6e6c8c # v2.5.0
+
+      - name: Docker build consul
+        run: docker build -t windows/consul -f Dockerfile-windows .
+
+      - name: Docker build consul local
+        shell: bash
+        run: cd build-support/windows && ./build-consul-local-images.sh
+
+      - name: Docker build consul dev
+        shell: bash
+        run: cd build-support/windows && ./build-consul-dev-image.sh
+
+      - name: Envoy Integration Tests for windows case-api-gateway-http-hostnames
```

Comment:
sure

Reply:
I tried but the matrix is producing more than 256 combinations - https://github.com/hashicorp/consul/actions/runs/5598185493 
So workflow is not triggering.
The reason tests are not run together in windows, is because they fail intermittently when run in groups separated by `|`. I discussed this with @huikang as well.

Reply:
What should be the next steps then @jmurret ?

Reply:
Can you link to the intermittent failures you experienced?  Before adding 1000 lines of CI code, we need to fully show the original approach did not work.  We should be running this just like the non-wondows tests and then documenting in comments if it does not work.  So I would love to see an example of intermittent failures that we might try to resolve before moving on to plan b, plan c, etc.  We clearly do this already, so if we take that time box a resolution to say 3 hours, we could possibly move it forward.  I'd rather not brainstorm on the other things before we close the door on the original approach that should match non-windows.

Reply:
Related to my other comment, if your team wants to move this PR forward and get tests running nightly, I think a sufficient resolution would be to add a task as a fast follow to unblock this PR.

My concerns with this is that is 1k extra lines where we have to remember to add new tests to each we add a test (IE the CI job will not pick it up automatically as it does in the non-windows version).  So if we want to move this PR forward, I would expect this to be a short lived solution with a fast follow task that is meant to be worked in on the next month.

Reply:
Some tests were failing randomly when we ran it in my EC2 Windows machine when grouped together. When run individually they were passing. It is not on CI. Also some tests we need to skip on windows like case-wasm, case-ingress-gateway-tls (because of windows support). Please tell me how to go about this?

Reply:
Will create a task for this.

Reply:
you can skip this in CI by piping `grep -v <test case name to exclude>`. 

Getting back to the original question, can you try the original approach and CI and document how it performs.  I would think if it passes in CI but flakes for you locally, it would be better to commit the original passing version of the CI configuration.  So before we backlog this, can you run the original approach in CI and link the results.  The plan b option of adding 1k lines to run each test individually should be avoided if possible, IMO.  We have no guarantees on backlog tasks ever getting worked on, so before we pass on this, it would be good to show it not working in CI and that being a blocker.

Reply:
Tests running together with job matrix - https://github.com/hashicorp/consul/actions/runs/5608653774/jobs/10261517975

Reply:
I am happy to move on from this if you and your team (@3loka) are aligned that this is getting separated out and what its priority is.

I am not trying to be an obstacle on this PR.  CI PRs are very time consuming, are huge, and take 300 runs to get right.  I just went through 2 months of nothing but migrating CI from CircleCI to GitHub actions.

Some things I can tell about helping the review process go faster:
- the reviewer will not know the ins and outs of every file, so for the most part they are checking:
  - does it work?
  - are all of the intended verifications accounted for?

So, as a PR author, I want to:
- make it easy for the reviewer to find proof this all works
  - links to where to verify this
- make it easy for the reviewer to see that all verifications are included and if there is not there is a plan for them
  - that includes commenting on why something is ommitted
  - also want the plan for it is
  - also mentioning that the team discussed this and agreed on this course of action
  - reviewers should know that the author is not just excluding stuff without team buy in

Working on CI can get very tiresome, so I appreciate and am impressed with your efforts.  I would ask that you add more detail related to the above when drafting PRs.  Also, time can obviously get lost in timezone differences.  So we should always be looking for ways to reduce the back and forth.  So, from any authoring side, if you could provide more thorough explanations in responses.  For example, a lot of your responses we short "It did not work in CI".  I think providing additional context in responses like what the error is, a link to the failure you saw, what you tried and where you gave up, etc. makes it easier on the reviewer to know again that everything has been thoroughly investigated in a time boxed manner and the author is aligned with their team on moving this forward and adding follow up tasks.  I'm open to any suggestions you have on the reviewing side to help lessen the time it takes to get to a common understanding!